core/src/OpenACC/Kokkos_OpenACC_DeepCopy.hpp

```diff
@@ -34,7 +34,7 @@ struct Kokkos::Impl::DeepCopy<Kokkos::Experimental::OpenACCSpace,
     // value checking is added as a safeguard. (The current NVHPC (V22.5)
     // supports OpenACC V2.7.)
     if (n > 0) {
-      acc_memcpy_device(dst, const_cast<void*>(src), n);
+      acc_memcpy_device_async(dst, const_cast<void*>(src), n, acc_async_noval);
```

Comment:
This overload of `DeepCopy`, without the leading execution space argument, is meant to be synchronous IIRC.

Reply:
We have fences at the call side, also see https://github.com/kokkos/kokkos/pull/4349.